### ISSUE TITLE
Crash when loading DDS with invalid format

### ISSUE THREAD

**Issue by ousnius** [g-truc/gli#118]
Users of our program are having problems with a specific DDS texture file with a format that doesn't seem to exist (uncompressed with a (255, 255, 255, 0) mask and, for some reason, the luminance bit turned on.

The Intel Photoshop plugin can't even load this specific texture file. Now, it would be fine if it was simply not loaded and an empty texture object was returned from the load_dds function, however it tries to create a texture object anyway but fails due to the invalid format (for example divisions by zero can happen).

Would it be possible to add something along those lines to the load_dds function, instead of  just the GLI_ASSERT macro for debug builds?
```
if (Format == static_cast<format>(gli::FORMAT_INVALID))
    return texture();
```

**Comment by repi** [comment]
Agree, one shouldn't verify file formats using asserts that are stripped out, We also ran into incorrect dds files that crashed in release and assert in debug, but just want all of those cases to be handled with returning an empty texture instead so we can catch it so we have local modifications here. Alternative support for throwing an exception instead on loading error, either would be more robust
